flax/linen/attention.py

```diff
@@ -287,10 +287,14 @@ def __call__(self,
     # Convert the boolean attention mask to an attention bias.
     if mask is not None:
       # attention mask in the form of attention bias
+      if self.dtype == jnp.float16:
+        large_compatible_negative = jnp.finfo(self.dtype).min
+      else:
+        large_compatible_negative = -1e10
       attention_bias = lax.select(
           mask > 0,
           jnp.full(mask.shape, 0.).astype(self.dtype),
-          jnp.full(mask.shape, -1e10).astype(self.dtype))
+          jnp.full(mask.shape, large_compatible_negative).astype(self.dtype))
```

Comment:
Could we make this change more concise by just adding a `jnp.clip` inline here instead of introducing a new long-winded variable name? e.g.
`jnp.clip(-1e10, jnp.finfo(self.dtype).min)`

(I'm just sensitive to trying not to let this layer code grow too much or add new names - there's already a lot of detail for a new reader to keep track of.)

Reply:
done.